Src/LiquidProjections.NHibernate/NHibernateProjector.cs

```diff
@@ -93,7 +98,20 @@ public string StateKey
 
             foreach (IList<Transaction> batch in transactions.InBatchesOf(BatchSize))
             {
-                await ProjectTransactionBatch(batch).ConfigureAwait(false);
+                Func<Task> projectTransactionBatchTask = async () => await ProjectTransactionBatch(batch);
+                try
+                {
+                    await projectTransactionBatchTask().ConfigureAwait(false);
+                }
+                catch (ProjectionException exception)
+                {
+                    if (ExceptionPolicy == null)

```

Comment:
If you set-up the default policy to throw, then you won't have to assert the policy has been set.